assets/src/edit-story/components/library/panes/pageTemplates/templateSave.js

```diff
@@ -80,38 +99,79 @@ const StyledText = styled(Text)`
   margin-bottom: 8px;
 `;
 
-const StyledButton = styled(Button)`
-  height: 32px;
-`;
+function TemplateSave({ pageSize, setShowDefaultTemplates, loadTemplates }) {
+  const {
+    actions: { addPageTemplate },
+  } = useAPI();
+  const { showSnackbar } = useSnackbar();
+
+  const { currentPage } = useStory(({ state: { currentPage } }) => ({
+    currentPage,
+  }));
 
-function TemplateSave({ pageSize }) {
   const { customPageTemplates } = useFeatures();
-  const handleSaveTemplate = useCallback(() => {}, []);
+  const handleSaveTemplate = useCallback(() => {
+    // Don't add empty page.
+    if (isDefaultPage(currentPage)) {
+      showSnackbar({
+        message: __(
+          'An empty page can’t be saved as a template. Add elements and try again.',
+          'web-stories'
+        ),
+        dismissable: true,
+      });
+    } else {
+      addPageTemplate({ ...currentPage, id: uuidv4(), title: null }).then(
```

Comment:
Nitpic, I find it clean use an async function here. See this example. 

https://github.com/google/web-stories-wp/blob/4818904a26b8574c6acfbd25a939fbdb0409c083/assets/src/edit-story/app/media/utils/useProcessVideo.js#L111-L126